### PR TITLE
feat: add more logging to schema manager

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

I was investigating integration tests, and realized that the Camunda Exporter was quite a black box. It was unclear what it is happening and doing on start up. this is the reason why added more extensive logging, which I think is beneficial especially on start up to understand what the Exporter is creating, updating, etc. This is also the reason why I put everything to info. We can discuss whether we want to put things like: `Create missing index` to debug though. Right now I find it useful, but this can easily changed anyway later as well.

Startup before:

```
21:38:13.309 [Broker-0] [ZeebePartition-1] [zb-actors-1] INFO  io.camunda.zeebe.broker.exporter - Use default exporter factory to create instance of class io.camunda.exporter.CamundaExporter
21:38:15.443 [Broker-0] [Exporter-1] [zb-fs-workers-1] INFO  io.camunda.exporter.CamundaExporter - Exporter opened
```

Startup now:
```
21:33:09.995 [Broker-0] [ZeebePartition-1] [zb-actors-0] INFO  io.camunda.zeebe.broker.exporter - Use default exporter factory to create instance of class io.camunda.exporter.CamundaExporter
21:33:10.023 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Schema creation is enabled. Start Schema management.
21:33:10.065 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Validate 0 existing indices based on 12 descriptors
21:33:10.394 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Validate 20 existing index templates based on 11 template descriptors
21:33:10.409 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing indices. [existing: 0, missing: 12]
21:33:10.410 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index identity-groups-8.7.0_
21:33:10.528 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index identity-role-8.7.0_
21:33:10.653 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index tasklist-form-8.4.0_
21:33:10.781 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index identity-tenants-8.7.0_
21:33:10.887 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index operate-decision-requirements-8.3.0_
21:33:10.996 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index identity-authorizations-8.7.0_
21:33:11.133 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index operate-metric-8.3.0_
21:33:11.314 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index operate-decision-8.3.0_
21:33:11.449 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index operate-process-8.3.0_
21:33:11.591 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index identity-users-8.7.0_
21:33:11.717 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index identity-mappings-8.7.0_
21:33:11.853 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index tasklist-metric-8.3.0_
21:33:12.014 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index templates. [existing: 26, missing: -15]
21:33:12.015 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index template tasklist-task-8.5.0_template
21:33:12.085 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Create missing index tasklist-task-8.5.0_ for template tasklist-task-8.5.0_template
21:33:12.240 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Update index schema. 0 indices need to be updated
21:33:12.240 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Update index template schema. 0 index templates need to be updated
21:33:12.240 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.schema.SchemaManager - Schema management completed.
21:33:12.245 [Broker-0] [Exporter-1] [zb-fs-workers-0] INFO  io.camunda.exporter.CamundaExporter - Exporter opened
```

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java [hunk 1]
```diff
@@ -89,13 +109,21 @@ private void initialiseIndexTemplates() {
             .getMappings(config.getIndex().getPrefix() + "*", MappingSource.INDEX_TEMPLATE)
             .keySet();
 
+    LOG.info(
+        "Create missing index templates. [existing: {}, missing: {}]",
+        existingTemplateNames.size(),
+        indexTemplateDescriptors.size() - existingTemplateNames.size());
```
Comment (EuroLew) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java:115]: This existing template names is always going to be the wrong size for an empty index prefix. It includes templates created by ES/OS such as `.monitoring-es`. If the prefix is always going to be set to a value this is irrelevant but if it defaults to an empty prefix then we can just add a `filter(name -> !name.startsWith(".")` should be good.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java [hunk 2]
```diff
@@ -143,24 +173,33 @@ private IndexSettings getIndexSettings(final String indexName) {
...
       return Map.of();
     }
 
     final var currentTemplates =
         searchEngineClient.getMappings(
             config.getIndex().getPrefix() + "*", MappingSource.INDEX_TEMPLATE);
 
+    LOG.info(
+        "Validate {} existing index templates based on {} template descriptors",
+        currentTemplates.size(),
```
Comment (EuroLew) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java:201]: same here

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java (hunk 1 above)
Comment (EuroLew) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java:115]: We could also change this in the `getMappings` to where we automatically filter out ES/OS created templates

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java [hunk 3]
```diff
@@ -65,22 +77,30 @@ public void initialiseResources() {
...
+        existingIndexNames.size(),
+        indexDescriptors.size() - existingIndexNames.size());
     indexDescriptors.stream()
         .filter(descriptor -> !existingIndexNames.contains(descriptor.getFullQualifiedName()))
         .forEach(
-            descriptor ->
-                searchEngineClient.createIndex(
-                    descriptor, getIndexSettings(descriptor.getIndexName())));
+            descriptor -> {
+              LOG.info("Create missing index {}", descriptor.getFullQualifiedName());
```
Comment (EuroLew) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java:95]: same here

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java [hunk 4]
```diff
@@ -89,13 +109,21 @@ private void initialiseIndexTemplates() {
...
     indexTemplateDescriptors.stream()
         .filter(descriptor -> !existingTemplateNames.contains(descriptor.getTemplateName()))
         .forEach(
             descriptor -> {
+              LOG.info("Create missing index template {}", descriptor.getTemplateName());
               searchEngineClient.createIndexTemplate(
                   descriptor, getIndexSettings(descriptor.getIndexName()), true);
-
+              LOG.info(
+                  "Create missing index {} for template {}",
```
Comment (EuroLew) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java:124]: same here surround name with []

**Review by EuroLew — APPROVED** [review]
LGTM, I'm ok with the log levels as is because it won't be too noisy due to schema startup not happening that often.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java (hunk 1 above)
Comment (ChrisKujawa) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java:115]: Could we also use a regex like `^[a-zA-Z]+.*` ? 🤔

Comment (EuroLew) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java:115]: that could work as well

**Comment by ChrisKujawa** [comment]
I just realized that we already have this? In the ES and Os clients? 🤔

**Comment by ChrisKujawa** [comment]
```
20:57:01.092 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [authorizations] was successfully created
20:57:01.165 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [groups] was successfully created
20:57:01.247 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [decision-requirements] was successfully created
20:57:01.320 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [decision] was successfully created
20:57:01.399 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [form] was successfully created
20:57:01.479 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [role] was successfully created
20:57:01.558 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [metric] was successfully created
20:57:01.634 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [mappings] was successfully created
20:57:01.736 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [process] was successfully created
20:57:01.816 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [metric] was successfully created
20:57:01.898 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [users] was successfully created
20:57:01.981 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [tenants] was successfully created
20:57:02.052 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Template [custom-prefix-operate-post-importer-queue-8.3.0_template] was successfully created
20:57:02.151 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [post-importer-queue] was successfully created
20:57:02.186 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Template [custom-prefix-tasklist-task-8.5.0_template] was successfully created
20:57:02.272 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [task] was successfully created
20:57:02.300 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Template [custom-prefix-operate-event-8.3.0_template] was successfully created
20:57:02.388 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Index [event] was successfully created
20:57:02.416 [] DEBUG io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient - Template [custom-prefix-tasklist-draft-task-variable-8.3.0_template] was successfully created

```

I just turned on debug loggin and saw this!

**Comment by ChrisKujawa** [comment]
I will merge this. We have a bit more overview in the schema manager like seeing counts etc. We might want to remove it from the clients, as it is also duplicate.

**Pull request merged**